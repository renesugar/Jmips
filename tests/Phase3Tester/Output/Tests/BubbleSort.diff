--- SelfTestCases/BubbleSort.out	2019-02-02 14:36:13.000000000 -0800
+++ ./Output/Tests/BubbleSort.out	2019-03-11 00:03:07.000000000 -0700
@@ -1,22 +1,5 @@
-20
-7
-12
-18
-2
-11
-6
-9
-19
-5
-99999
-2
-5
-6
-7
-9
-11
-12
-18
-19
-20
-0
+--- Error (ProgramFault) ---
+require data address, found integer44
+  Main() line 17
+  BBS.Start() line 29
+  BBS.Init() line 185
