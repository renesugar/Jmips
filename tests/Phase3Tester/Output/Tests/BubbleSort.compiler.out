const vmt_BBS
  :BBS.Start
  :BBS.Sort
  :BBS.Print
  :BBS.Init

func Main [in 0, out 0, local 0]
  $t8 = HeapAllocZ(12)
  [$t8] = :vmt_BBS
  if $t8 goto :null1
  Error("null pointer")
null1:
  $t7 = [$t8]
  $t7 = [$t7]
  $a0 = $t8
  $a1 = 10
  call $t7
  $t6 = $v0
  PrintIntS($t6)
  ret

func BBS.Start [in 0, out 0, local 0]
  $t7 = $a0
  $t6 = $a1
  $t8 = [$t7]
  $t8 = [$t8+12]
  $a0 = $t7
  $a1 = $t6
  call $t8
  $t5 = $v0
  $t8 = [$t7]
  $t8 = [$t8+8]
  $a0 = $t7
  call $t8
  $t5 = $v0
  PrintIntS(99999)
  $t6 = [$t7]
  $t6 = [$t6+4]
  $a0 = $t7
  call $t6
  $t5 = $v0
  $t8 = [$t7]
  $t8 = [$t8+8]
  $a0 = $t7
  call $t8
  $t5 = $v0
  $v0 = 0
  ret

func BBS.Sort [in 0, out 0, local 0]
  $t8 = $a0
  $t7 = [$t8+8]
  $t6 = Sub($t7 1)
  $t7 = Sub(0 1)
while1_top:
  $t5 = LtS($t7 $t6)
  if0 $t5 goto :while1_end
  $t7 = 1
while2_top:
  $t5 = Add($t6 1)
  $t4 = LtS($t7 $t5)
  if0 $t4 goto :while2_end
  $t5 = Sub($t7 1)
  $t4 = [$t8+4]
  if $t4 goto :null2
  Error("null pointer")
null2:
  $t3 = [$t4]
  $t3 = Lt($t5 $t3)
  if $t3 goto :bounds1
  Error("array index out of bounds")
bounds1:
  $t3 = MulS($t5 4)
  $t3 = Add($t3 $t4)
  $t5 = [$t3+4]
  $t4 = [$t8+4]
  if $t4 goto :null3
  Error("null pointer")
null3:
  $t3 = [$t4]
  $t3 = Lt($t7 $t3)
  if $t3 goto :bounds2
  Error("array index out of bounds")
bounds2:
  $t3 = MulS($t7 4)
  $t3 = Add($t3 $t4)
  $t4 = [$t3+4]
  $t3 = LtS($t4 $t5)
  if0 $t3 goto :if1_else
  $t5 = Sub($t7 1)
  $t4 = [$t8+4]
  if $t4 goto :null4
  Error("null pointer")
null4:
  $t3 = [$t4]
  $t3 = Lt($t5 $t3)
  if $t3 goto :bounds3
  Error("array index out of bounds")
bounds3:
  $t3 = MulS($t5 4)
  $t3 = Add($t3 $t4)
  $t4 = [$t3+4]
  $t3 = [$t8+4]
  if $t3 goto :null5
  Error("null pointer")
null5:
  $t2 = [$t3]
  $t2 = Lt($t5 $t2)
  if $t2 goto :bounds4
  Error("array index out of bounds")
bounds4:
  $t2 = MulS($t5 4)
  $t2 = Add($t2 $t3)
  $t5 = [$t8+4]
  if $t5 goto :null6
  Error("null pointer")
null6:
  $t3 = [$t5]
  $t3 = Lt($t7 $t3)
  if $t3 goto :bounds5
  Error("array index out of bounds")
bounds5:
  $t3 = MulS($t7 4)
  $t3 = Add($t3 $t5)
  $t5 = [$t3+4]
  [$t2] = $t5
  $t5 = [$t8+4]
  if $t5 goto :null7
  Error("null pointer")
null7:
  $t3 = [$t5]
  $t3 = Lt($t7 $t3)
  if $t3 goto :bounds6
  Error("array index out of bounds")
bounds6:
  $t3 = MulS($t7 4)
  $t3 = Add($t3 $t5)
  [$t3] = $t4
  goto :if1_end
if1_else:
  $t8 = 0
if1_end:
  $t7 = Add($t7 1)
  goto :while2_top
while2_end:
  $t6 = Sub($t6 1)
  goto :while1_top
while1_end:
  $v0 = 0
  ret

func BBS.Print [in 0, out 0, local 0]
  $t6 = $a0
  $t8 = 0
while3_top:
  $t7 = [$t6+8]
  $t5 = LtS($t8 $t7)
  if0 $t5 goto :while3_end
  $t7 = [$t6+4]
  if $t7 goto :null8
  Error("null pointer")
null8:
  $t5 = [$t7]
  $t5 = Lt($t8 $t5)
  if $t5 goto :bounds7
  Error("array index out of bounds")
bounds7:
  $t5 = MulS($t8 4)
  $t5 = Add($t5 $t7)
  $t6 = [$t5+4]
  PrintIntS($t6)
  $t8 = Add($t8 1)
  goto :while3_top
while3_end:
  $v0 = 0
  ret

func BBS.Init [in 0, out 0, local 0]
  $t7 = $a0
  $t8 = $a1
  [$t7] = $t8
  $a0 = $t8
  call :AllocArray
  $t6 = $v0
  [$t7] = $t6
  $t8 = [$t7+4]
  if $t8 goto :null9
  Error("null pointer")
null9:
  $t6 = [$t8]
  $t6 = Lt(0 $t6)
  if $t6 goto :bounds8
  Error("array index out of bounds")
bounds8:
  $t6 = MulS(0 4)
  $t6 = Add($t6 $t8)
  [$t6] = 20
  $t8 = [$t7+4]
  if $t8 goto :null10
  Error("null pointer")
null10:
  $t6 = [$t8]
  $t6 = Lt(1 $t6)
  if $t6 goto :bounds9
  Error("array index out of bounds")
bounds9:
  $t6 = MulS(1 4)
  $t6 = Add($t6 $t8)
  [$t6] = 7
  $t8 = [$t7+4]
  if $t8 goto :null11
  Error("null pointer")
null11:
  $t6 = [$t8]
  $t6 = Lt(2 $t6)
  if $t6 goto :bounds10
  Error("array index out of bounds")
bounds10:
  $t6 = MulS(2 4)
  $t6 = Add($t6 $t8)
  [$t6] = 12
  $t8 = [$t7+4]
  if $t8 goto :null12
  Error("null pointer")
null12:
  $t6 = [$t8]
  $t6 = Lt(3 $t6)
  if $t6 goto :bounds11
  Error("array index out of bounds")
bounds11:
  $t6 = MulS(3 4)
  $t6 = Add($t6 $t8)
  [$t6] = 18
  $t8 = [$t7+4]
  if $t8 goto :null13
  Error("null pointer")
null13:
  $t6 = [$t8]
  $t6 = Lt(4 $t6)
  if $t6 goto :bounds12
  Error("array index out of bounds")
bounds12:
  $t6 = MulS(4 4)
  $t6 = Add($t6 $t8)
  [$t6] = 2
  $t8 = [$t7+4]
  if $t8 goto :null14
  Error("null pointer")
null14:
  $t6 = [$t8]
  $t6 = Lt(5 $t6)
  if $t6 goto :bounds13
  Error("array index out of bounds")
bounds13:
  $t6 = MulS(5 4)
  $t6 = Add($t6 $t8)
  [$t6] = 11
  $t8 = [$t7+4]
  if $t8 goto :null15
  Error("null pointer")
null15:
  $t6 = [$t8]
  $t6 = Lt(6 $t6)
  if $t6 goto :bounds14
  Error("array index out of bounds")
bounds14:
  $t6 = MulS(6 4)
  $t6 = Add($t6 $t8)
  [$t6] = 6
  $t8 = [$t7+4]
  if $t8 goto :null16
  Error("null pointer")
null16:
  $t6 = [$t8]
  $t6 = Lt(7 $t6)
  if $t6 goto :bounds15
  Error("array index out of bounds")
bounds15:
  $t6 = MulS(7 4)
  $t6 = Add($t6 $t8)
  [$t6] = 9
  $t8 = [$t7+4]
  if $t8 goto :null17
  Error("null pointer")
null17:
  $t6 = [$t8]
  $t6 = Lt(8 $t6)
  if $t6 goto :bounds16
  Error("array index out of bounds")
bounds16:
  $t6 = MulS(8 4)
  $t6 = Add($t6 $t8)
  [$t6] = 19
  $t8 = [$t7+4]
  if $t8 goto :null18
  Error("null pointer")
null18:
  $t6 = [$t8]
  $t6 = Lt(9 $t6)
  if $t6 goto :bounds17
  Error("array index out of bounds")
bounds17:
  $t6 = MulS(9 4)
  $t6 = Add($t6 $t8)
  [$t6] = 5
  $v0 = 0
  ret

func AllocArray [in 0, out 0, local 0]
  $t8 = $a0
  $t7 = MulS($t8 4)
  $t7 = Add($t7 4)
  $t6 = HeapAllocZ($t7)
  [$t6] = $t8
  $v0 = $t6
  ret


